pkg/nodeagent/controller/operatingsystemconfig/changes.go

```diff
@@ -68,6 +70,16 @@ type files struct {
 	deleted []extensionsv1alpha1.File
 }
 
+type containerd struct {
+	configChange bool
+	registries   containerdRegistries
+}
+
+type containerdRegistries struct {
+	current []extensionsv1alpha1.RegistryConfig
```

Comment:
Nit: `desired` could be a better name I am also ok if you decide to leave it as it is

```suggestion
	desired []extensionsv1alpha1.RegistryConfig
```